### ISSUE TITLE
Games from Steam Families not appearing in Lutris

### ISSUE THREAD

**Issue by bjoern-tantau** [lutris/lutris#5393]
I don't know if it was the same with the old Steam family sharing. But in the new Steam Families the games you get through your family are not shown on the profile page and thus also not in Lutris. Even if you have them installed.

But the Families games do show up in the last played section of the profile. Maybe that could be a workaround to get them into Lutris.

Of course it would be nicer if all the games were showing in Lutris.

**Comment by Gab0-here** [comment]
I have the same issue, most of the games I play are on the steam family. Even if I can open steam and play them from there, it would definitely be better if it was integrated within Lutris.

It does make sense why it can't, since it's an opt-in option that has to be configured on every single client, and I don't even know what API they use, or if it can be activated there.

With that said you can still add them manually, and they WILL get recognized and work, you just need to add them as "locally installed game", select STEAM as the launcher, and put the steam app ID (for example, 1091500 for cyberpunk 2077), if you don't know the ID or how to get it, you can search for it on SteamDB.

![Screenshot from 2024-07-12 23-31-47](https://github.com/user-attachments/assets/c266bf4f-1b02-43a8-929b-a546588c7b54)

![Screenshot from 2024-07-12 23-32-01](https://github.com/user-attachments/assets/a3489570-0da8-410d-8414-818f7e34422c)

![Screenshot from 2024-07-12 23-33-29](https://github.com/user-attachments/assets/6573ac93-550c-488b-a1c9-99bab8722b61)

![Screenshot from 2024-07-12 23-33-38](https://github.com/user-attachments/assets/df4ad4c2-50cd-4c55-a0b5-08af83647a27)

**Comment by voidlesity** [comment]
Any updates on this?
It would be really cool to have installed family games shown in lutris

**Comment by GamerWierdo100** [comment]
I too have this issue

**Comment by Alphenex** [comment]
Same issue, this definitely should be fixed/added

**Comment by AliensInc** [comment]
I don't use the latest release, I use the git-version and this work.
Yesterday I installed a couple of games from steam and they all was there.

Download https://github.com/lutris/lutris and then run the bin/lutris script, do not install.

If this works then it has been fixed in the code(not yet released), if not there gotta be something wrong with your installation.

Just my thoughts.

**Comment by bjoern-tantau** [comment]
> I don't use the latest release, I use the git-version and this work. Yesterday I installed a couple of games from steam and they all were there.
> 
> Download https://github.com/lutris/lutris and then run the bin/lutris script, do not install.
> 
> If this works then it has been fixed in the code(not yet released), if not there gotta be something wrong with your installation.
> 
> Just my thoughts.

This isn't about installed games. Games a family member has don't appear in the list of available games.

**Comment by AliensInc** [comment]
Uh, do you mean another account?

also what I mean isn't just the installed games but all games, installed and not insralled that I have on this particular account... and that is what I thought you meant.

**Comment by bjoern-tantau** [comment]
Yes, with the way the new Steam Family works you get all the games of all family members. But Lutris shows just your own games.

Example:

* you have Goat Simulator
* your son has Doki Doki Literature Club
* your daughter has Doom Eternal

Steam shows that you have Goat Simulator, Doki Doki Literature Club and Doom Eternal. But Lutris will only show Goat Simulator. If you install Doki Doki Literature Club Lutris will also show that game.

**Comment by AliensInc** [comment]
OK, never knew this was supposed to be a feature, to show other accounts games.

Ah well, it's just me on my machines and got no kids(thank god😜) so I can't test this other than to create another steam account......
Uh wait... or do you mean SAME steam  account but different Lutris account?
If this is a thing(?)

Anywho, I'm out cos I can't help ya 😥

**Comment by rosxz** [comment]
I haven't ever touched or looked at any Lutris code before, nor any Steam APIs, however when trying the endpoint Steam WebAPI endpoint **IFamilyGroupsService/GetSharedLibraryApps** with the API Key, I always get "Unauthorized error ... Please verify your key= parameter".
Checking how Playnite is going, it seems to me it's [not just a skill issue on my end](https://github.com/JosefNemec/PlayniteExtensions/issues/396#issuecomment-2246940318), and Lutris would need to retrieve the user's Steam store access token (which also has a short lifespan). Trying the same endpoint with the access token works on my end.
However, to get the access token it would probably mean adding a similar login prompt as with Epic Games but for Steam, and have it automatically refresh the token once it expires. I made a quick search for methods of interacting with the native steam client (as an alternative), but I didn't find anything relevant.

**Comment by rosxz** [comment]
I added the Steam family shared games in this [fork](https://github.com/rosxz/lutris). It's rough as is, but it does the job. 

Add the Steam Family source and login to your account, then the games should be added to the Steam source (I don't know why they aren't going to the steam family source 😄 ). The games also seem to be classified as generic Linux games, not as Steam games, as the Steam icon for family games doesn't appear near the title. Anyway I don't recommend using rn it if you really care about all that.

I might continue fixing it next week or so.

**Comment by danieljohnson2** [comment]
That might make a good PR with a bit of polish, but there's a bunch of extra stuff you've committed- ``default.nix``, ``flake.lock`` that sort of thing. You'll need to separate that out.

**Comment by rosxz** [comment]
> That might make a good PR with a bit of polish, but there's a bunch of extra stuff you've committed- `default.nix`, `flake.lock` that sort of thing. You'll need to separate that out.

@danieljohnson2 I went ahead and removed those files and fixed the issues I could find. It can now properly deal with token expiry and the games are in the correct source section. I added a settings value STEAM_FAMILY_INCLUDE_OWN in case the users prefer to have every Steam game under one Steam section (it is turned off by default).

Can I open a PR?

![Image](https://github.com/user-attachments/assets/d1996f92-55e4-46d1-af5d-69d0e458cfc8)

**Comment by danieljohnson2** [comment]
Sure, I think you can submit a PR. I'll let the boss decide if he wants this, but there's certainly demand for it out there.

**Comment by OriCat101** [comment]
Any updates on this? The PR got merged but I cannot find the Source in the latest release

**Comment by Zebra2711** [comment]
> Any updates on this? The PR got merged but I cannot find the Source in the latest release

You can test with this build:
https://github.com/Zebra2711/lutris/releases/tag/v0.5.19

Also, i recommend using the lutris flatpak, which is made by the official lutris team

**Comment by OriCat101** [comment]
> You can test with this build:
https://github.com/Zebra2711/lutris/releases/tag/v0.5.19

So I pointed the nixpkg to there and I still dont see it and I wanned to ask if it is only included in the build or also in the repo?

EDIT: managed to get it working by using lutris-unwrapped
```nix
let
  steamFamilyLutris = pkgs.lutris-unwrapped.overrideAttrs (oldAttrs: {
    src = pkgs.fetchFromGitHub {
      owner = "Zebra2711";
      repo = "lutris";
      rev = "v0.5.19";  # e.g., "main" or a specific commit hash
      sha256 = "sha256-1DI/cOIIYQhvArCx659rvnzWqZKK1Qhmv96Uq5mCJ3E=";  # Replace with actual hash (run with dummy first, Nix will suggest the correct one)
    };
  });
in
{
  environment.systemPackages = [ steamFamilyLutris ];
}
```

**Comment by Zebra2711** [comment]
tag v0.5.19 does not contain MR [#5968 Add support for Steam Family Sharing ](https://github.com/lutris/lutris/pull/5968)

**Comment by GamerWierdo100** [comment]
@Zebra2711 What update does have that merge? I'm using archlinux latest but I can't seem to see it.
